cluster-operator/src/main/java/io/strimzi/operator/cluster/Main.java

```diff
@@ -121,11 +117,11 @@ static CompositeFuture run(Vertx vertx, KubernetesClient client, PlatformFeature
                 networkPolicyOperator, podDisruptionBudgetOperator, resourceOperatorSupplier, config.versions(), config.getImagePullPolicy());
 
         KafkaConnectS2IAssemblyOperator kafkaConnectS2IClusterOperations = null;
-        if (pfa.isOpenshift()) {
+        if (pfa.isBuilds() && pfa.isApps() && pfa.isImages()) {
             kafkaConnectS2IClusterOperations = createS2iOperator(vertx, client, pfa, serviceOperations,
                     configMapOperations, secretOperations, certManager, resourceOperatorSupplier, config.versions(), config.getImagePullPolicy());
         } else {
-            maybeLogS2iOnKubeWarning(vertx, client);
+            log.warn("The KafkaConnectS2I custom resource definition can only be used in environment which supports OpenShift build, image and apps APIs. These APIs do not seem to be supported in this environment.");
```

Comment:
I think the old way of doing this (where we check that the S2I CRD is not installed on a platform which doesn't support it) is better. As this is currently written I think this will produce a warning on kube every time the operator starts, which doesn't give a good impression.

Reply:
TBH, the old way never worked because it never had the RBAC. But because it had empty `catch` block which just hid the error. So it actually never gave the message. I'm also not convinced by the whole idea of deleting the CRD. It can mess up with the installations using OLM / OperatorHub.io etc.

But if you insist on using this I can of course get it back and fix the RBAC.

Reply:
No I don't insist. I just think printing a warn every time just because we're running on kube is suboptimal. To me it should just an info at this point. If the operator later observes an s2i CR _that_ should be a warning. (And if we can't set up the watch on s2i because the CRD doesn't exist then that's fine.)

Reply:
I'm fine to change that to INFO level only. I think it's better to write it at startup then get yet another RBAC permissions for this. When the OpenShfit features are not supported, we actually don't setup the watch for the custom resource. So we will not know about them when someone creates them.

This is something I was looking into, but because of the OpenShift client and how we create the resource operators in `Main`, it would be quite a big refactoring to keep the watch even on Kubernetes and throw an error only when someone actually creates the resource.